VogtObserver3/Store.cs

```diff
@@ -9,22 +9,34 @@ class Store : IObserveable
 
         private readonly List<Observer> _observers = new List<Observer>();
 
+        private float _discount;
+
         public void AddObserver(Observer o)
         {
             _observers.Add(o);
         }
 
-        public void Notify()
+        public void Notify(float discount)
         {
             foreach (var observer in _observers)
             {
-
+                observer.Update(discount);
             }
         }
 
         public void RemoveObserver(Observer o)
         {
             _observers.Remove(o);
         }
+
+
+        public void ApplyDiscount()
+        {
+            Random random = new Random();
```

Comment:
Less important, but it still sets off my OCD. I'd expect there to be 1 random object per class, and in many cases.. 1 random object per application.

Easy enough to make this class level.

```C#
class Customer : IObservable {
  private Random _random = new Random();
}
```

Then you aren't creating this new object over and over.

Reply:
Remember! `new .....` means its reserving memory for that object. And if you stop using the object, it has to "clean" it up. Creating the same new object over and over like this is a code smell.